src/mscorlib/shared/System/Globalization/DateTimeFormatInfo.cs

```diff
@@ -460,35 +454,12 @@ public static DateTimeFormatInfo CurrentInfo
             }
         }
 
-
-        public static DateTimeFormatInfo GetInstance(IFormatProvider provider)
-        {
-            // Fast case for a regular CultureInfo
-            DateTimeFormatInfo info;
-            CultureInfo cultureProvider = provider as CultureInfo;
-            if (cultureProvider != null && !cultureProvider._isInherited)
-            {
-                return cultureProvider.DateTimeFormat;
-            }
-            // Fast case for a DTFI;
-            info = provider as DateTimeFormatInfo;
-            if (info != null)
-            {
-                return info;
-            }
-            // Wasn't cultureInfo or DTFI, do it the slower way
-            if (provider != null)
-            {
-                info = provider.GetFormat(typeof(DateTimeFormatInfo)) as DateTimeFormatInfo;
-                if (info != null)
-                {
-                    return info;
-                }
-            }
-            // Couldn't get anything, just use currentInfo as fallback
-            return CurrentInfo;
-        }
-
+        public static DateTimeFormatInfo GetInstance(IFormatProvider provider) =>
+            provider == null ? CurrentInfo :
+            provider is CultureInfo cultureProvider && !cultureProvider._isInherited ? cultureProvider.DateTimeFormat :
```

Comment:
>!cultureProvider._isInherited [](start = 55, length = 29)

This check is interesting to the Desktop only. I am ok to keep it for app compat but it is really not needed for .net core.